lib/puppet/ssl/state_machine.rb

```diff
@@ -280,14 +280,11 @@ def next_state
 
         # close persistent connections and session state before sleeping
         Puppet.runtime['http'].close
-        @machine.session = nil
+        @machine.session = Puppet.runtime['http'].create_session
 
+        @machine.unlock
         Kernel.sleep(time)
-
-        # our ssl directory may have been cleaned while we were
-        # sleeping, start over from the top
-        @machine.session = Puppet.runtime['http'].create_session
```

Comment:
Does the risk of the ssl directory being cleaned still apply? Why move creating a new session to before we sleep?

Reply:
Ah yeah, that comment is really about moving to the NeedCACerts state after sleeping, so we either load the certs (if they're still there) or download them. We also want to clear out any resolver state for the session, like which server_list entry to use. I moved the session up just to reduce the code, as there's no harm in creating an empty session before sleeping.